### PR TITLE
Revamp the Welcome program

### LINKED ISSUE #1211: Welcome: crash on startup
**Reproduction:**
1. Open Welcome

**Stacktrace:**
```
USERSPACE(24) ASSERTION FAILED: start + length <= m_impl->length()
../../AK/String.cpp:121 in AK::String AK::String::substring(size_t, size_t) const
terminate_due_to_signal Welcome(24) <- 6
[Finalizer(3:3)]: 0xdeadc0de  Scheduler::switch_now() +63
0xdeadc0de  Thread::die_if_needed() +120
0xdeadc0de  syscall_handler +926
0xdeadc0de  syscall_asm_entry +28
0x080aae8e  raise +22
0x080482fd  abort +13
0x080aae16  kill +0
0x0809b378  AK::String::substring(unsigned long, unsigned long) const +84
0x0804a1cf  TextWidget::wrap_and_set_height() +291
0x0804a4e5  TextWidget::set_text(AK::StringView const&) +133
0x08048e66  main +2749
0x08049723  _start +94
```

### PR DESCRIPTION
**Pull request by thatlittlegit**
The Welcome program previously crashed due to an AK assertion, and fixing that let it start but had the word-wrapping mess up. This PR fixes those issues, and also changes the way the program works.

To begin, the PNG was switched out for a gradient instead. That way, it can deal better with window size changes, instead of staying the same size despite the window. I also had the buttons be pushed in when the page is selected, to make it more obvious which page you're on, switched the 'Welcome to Serenity' text to be in a heading font, and added pledge/unveil.

The largest overall change is moving the contents of the pages from the C++ source code into a text file inside /res. This also came with some other changes, like allowing icons.

Some things in here that I think might need to be changed elsewhere, as the code seems a bit short to have exclusive to Welcome:

* UnuncheckableButton: only exists because GUI::AbstractButton and GUI::Button don't allow changing the uncheckability. Should this be made a getter/setter on that instead?
* BackgroundWidget: exists because a gradient cannot be set as the background for a window, and it seems to be a bit tricky to get a custom color background for a widget. Should this be a GUI::Window thing instead?
* TextWidget: works fine, but should it be moved into LibGUI? It is quite generic...

This PR does fix #1211.

### CODE REVIEW

**Comment by awesomekling** [comment]
Thank you for working on this, it looks very tidy! :)

* Please use your own name for the copyright header in new files you add :) If you prefer not to use your name, you can put something generic like "The SerenityOS developers" instead.

* Uncheckability could be an attribute of `GUI::Button` instead of a separate class like you suggest.

* Going forward we should replace `TextWidget` with `LibHTML`, possibly together with `LibMarkdown`

**Comment by awesomekling** [comment]
The way to set a background color on a widget is via the palette system:

```cpp
auto pal = widget->palette();
pal.set_color(Gfx::ColorRole::Window, my_color);
widget->set_palette(pal);
```

**Comment by thatlittlegit** [comment]
I fixed the copyright notice. If I work on GUI::Button, I'll do it in a separate PR (since it isn't really related to Welcome).

The palette system also makes sense when I see it that way! At first I thought it'd change the colors across the entire system, which doesn't make it very helpful for single-window changes.

File: Applications/Welcome/main.cpp [hunk 1]
```diff
@@ -129,21 +192,35 @@ int main(int argc, char** argv)
...
         content->set_size_policy(GUI::SizePolicy::Fill, GUI::SizePolicy::Fill);
 
-        auto content_title = GUI::Label::construct(content);
+        auto title_box = GUI::Widget::construct(content.ptr());
+        title_box->set_layout(make<GUI::HorizontalBoxLayout>());
+        title_box->layout()->set_spacing(4);
+        title_box->set_preferred_size(0, 16);
+        title_box->set_size_policy(GUI::SizePolicy::Fill, GUI::SizePolicy::Fixed);
+
+        if (page.icon != String::empty()) {
```
Comment (shannonbooth) [inline Applications/Welcome/main.cpp]: maybe `!page.icon.is_empty()` instead? (this is in a few places)

File: Applications/Welcome/main.cpp [hunk 2]
```diff
@@ -24,94 +24,157 @@
...
+    StringBuilder current_output_line;
+    bool started = false;
+    ContentPage current;
+    while (true) {
+        auto buffer = file->read_line(4096);
+        if (buffer.is_null()) {
+            if (file->error()) {
+                file->close();
+                return error;
+            } else {
```
Comment (shannonbooth) [inline Applications/Welcome/main.cpp]: removing this else-after-return would slightly reduce indentation

**Comment by awesomekling** [comment]
Hmm, I tried fixing the merge conflicts using the GitHub web editor but apparently that was not enough. Could you rebase your changes on top of master and then I can merge this cleanly :)

**Comment by thatlittlegit** [comment]
Is there anything more I need to do?

**Comment by awesomekling** [comment]
Looks good! Thanks for working on this :)

**Comment by thatlittlegit** [comment]
Thank you!

**Pull request merged**